File: problems/0021_Merge_Two_Sorted_Lists/Bruce0021.py
# 21题，混合2链表并排序
class Solution:
    def mergeTwoLists(self, a, b):
        if not a or b and a.val > b.val:
	        a, b = b, a
	    if a:
	        a.next = self.mergeTwoLists(a.next, b)
	    return a
	
"""
# solution 2 还没看懂啊！20190313
class Solution:
    def mergeTwoLists(self, a, b):
	    if a and b:
		    if a.val > b.val:
			    a, b = b, a
			a.next = self.mergeTwoLists(a.next, b)
		return a or b
		
"""	
	

File: problems/0021_Merge_Two_Sorted_Lists/Bruce0021.cpp
//Bruce0021.cpp
// 混合插入有序链表


class Solution {

public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode *dummy = new ListNode(-1), 
        *cur = dummy;
        
        while (l1 && l2) {
            if (l1->val < l2->val) {
                cur->next = l1;
                l1 = l1->next;
            } else {
                cur->next = l2;
                l2 = l2->next;
            }
            cur = cur->next;
        }
        cur->next = l1 ? l1 : l2;
        return dummy->next;
    }
};